server/methods.js

```diff
@@ -0,0 +1,22 @@
+ Meteor.startup(function() {
+  if(Meteor.isServer) {
+    fuzzy = Npm.require("fuzzy");
+  }
+});
+
+ Meteor.methods({
+  findDreamCollegeId: function(userInputCollege) {
+    check(userInputCollege, String);
+    var colleges = CollegesInMemory;
+    var collegeNameArray = [];
+    var collegeNameHash = {}
+    _.each(colleges, function(college) {
+      collegeNameArray.push(college.name);
+      collegeNameHash[college.name] = college._id;
+    })
+    var results = fuzzy.filter(userInputCollege, collegeNameArray);
```

Comment:
A Mongo $text search is probably fine. The only issue I have with text searches is that they work on a per-word level. I don't think "Berkley" would return a college with a name of "Berkeley" using a $text search, but I don't think this is a huge issue.